exchanges/huobi/huobi.go

```diff
@@ -64,6 +175,1637 @@ type HUOBI struct {
+}
+
+// GetSwapMarketOverview gets market data overview for perpetual futures
+func (h *HUOBI) GetSwapMarketOverview(code string) (MarketOverviewData, error) {
+	var resp MarketOverviewData
+	params := url.Values{}
+	params.Set("contract_code", code)
+	return resp, h.SendHTTPRequest(exchange.RestFutures, huobiMarketDataOverview+params.Encode(), &resp)
+}
+
+// GetLastTrade gets the last trade for a given perpetual contract
+func (h *HUOBI) GetLastTrade(code string) (LastTradeData, error) {
+	var resp LastTradeData
+	params := url.Values{}
+	params.Set("contract_code", code)
+	return resp, h.SendHTTPRequest(exchange.RestFutures, huobiLastTradeContract+params.Encode(), &resp)
+}
+
+// GetBatchTrades gets batch trades for a specified contract (fetching size cannot be bigger than 2000)
+func (h *HUOBI) GetBatchTrades(code string, size int64) (BatchTradesData, error) {
+	var resp BatchTradesData
+	params := url.Values{}
+	params.Set("contract_code", code)
+	if !(size > 1) && !(size < 2000) {
+		return resp, fmt.Errorf("invalid size")
+	}
+	params.Set("size", strconv.FormatInt(size, 10))
+	return resp, h.SendHTTPRequest(exchange.RestFutures, huobiRequestBatchOfTradingRecords+params.Encode(), &resp)
+}
+
+// GetInsuranceData gets insurance fund data and clawback rates
+func (h *HUOBI) GetInsuranceData(code string) (InsuranceAndClawbackData, error) {
+	var resp InsuranceAndClawbackData
+	params := url.Values{}
+	params.Set("contract_code", code)
+	return resp, h.SendHTTPRequest(exchange.RestFutures, huobiInsuranceBalanceAndClawbackRate+params.Encode(), &resp)
+}
+
+// GetHistoricalInsuranceData gets historical insurance fund data and clawback rates
+func (h *HUOBI) GetHistoricalInsuranceData(code string, pageIndex, pageSize int64) (HistoricalInsuranceFundBalance, error) {
+	var resp HistoricalInsuranceFundBalance
+	params := url.Values{}
+	params.Set("contract_code", code)
+	if pageIndex != 0 {
+		params.Set("page_index", strconv.FormatInt(pageIndex, 10))
+	}
+	if pageSize != 0 {
+		params.Set("page_size", strconv.FormatInt(pageIndex, 10))
+	}
+	return resp, h.SendHTTPRequest(exchange.RestFutures, huobiInsuranceBalanceHistory+params.Encode(), &resp)
+}
+
+// GetTieredAjustmentFactorInfo gets tiered adjustment factor data
+func (h *HUOBI) GetTieredAjustmentFactorInfo(code string) (TieredAdjustmentFactorData, error) {
+	var resp TieredAdjustmentFactorData
+	params := url.Values{}
+	params.Set("contract_code", code)
+	return resp, h.SendHTTPRequest(exchange.RestFutures, huobiTieredAdjustmentFactor+params.Encode(), &resp)
+}
+
+// GetOpenInterestInfo gets open interest data
+func (h *HUOBI) GetOpenInterestInfo(code, period, amountType string, size int64) (OpenInterestData, error) {
+	var resp OpenInterestData
+	params := url.Values{}
+	params.Set("contract_code", code)
+	if !common.StringDataCompare(validPeriods, period) {
+		return resp, fmt.Errorf("invalid period value received")
+	}
+	params.Set("period", period)
+	if !(size > 0 && size <= 1200) {
```

Comment:
This can be simplified; size is used in other functions but should error the same as this one. All the others need to be updated.